utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -297,7 +303,7 @@ def generate_baseline(self):
         basegen_dir = os.path.join(baselineroot, self._case.get_value("BASEGEN_CASE"))
         for bdir in (baselineroot, basegen_dir):
             if not os.path.isdir(bdir):
-                append_status("GFAIL %s baseline\n" % self._case.get_value("CASEBASEID"),
+                append_status("FAIL %s generate\n" % self._case.get_value("CASEBASEID"),
```

Comment:
Dealing with NLFAIL was a pain. I'd prefer fewer status types rather than more. The user can always get more info form TestStatus.log.